src/plugins/URLIndicatorPlugin/URLIndicatorSpec.js

```diff
@@ -21,138 +21,166 @@
-                );
-            });
-            it("has a custom label if supplied in initialization", function () {
-                opts = {
-                    url: "http://localhost:8080",
-                    label: "Localhost"
-                };
-                indicatorWrapper = new Indicator();
-                expect(indicatorWrapper.getText()).toEqual("Checking status of Localhost please stand by...");
-            });
-            it("has a default label if not supplied in initialization", function () {
-                opts = {
-                    url: "http://localhost:8080"
-                };
-                indicatorWrapper = new Indicator();
-                expect(indicatorWrapper.getText()).toEqual(
-                  "Checking status of http://localhost:8080 please stand by..."
-                );
-            });
-            it("has a default interval if not supplied in initialization", function () {
-                opts = {
-                    url: "http://localhost:8080"
-                };
-                indicatorWrapper = new Indicator();
-                expect(mockInterval).toHaveBeenCalledWith(
-                    jasmine.any(Function),
-                    10000,
-                    0,
-                    false
-                );
+                    waitsFor(httpRequestReturned);
+                    runs(function () {
+                        var iconElement = getIconElement();
+                        expect(iconElement.hasClass('s-status-ok')).toBe(true);
+                    });
+                });
+
+                it("indicates an error when the server cannot be reached", function () {
+                    mockError();
+                    mockInterval(options.interval + 1);
+
+                    waitsFor(httpRequestReturned);
+                    runs(function () {
+                        var iconElement = getIconElement();
+                        expect(iconElement.hasClass('s-status-warning-hi')).toBe(true);
+                    });
+                });
             });
+
+            function mockHttpRequest() {
+                mockHttpRequestFunction = jasmine.createSpy('get');
+                URLIndicator.prototype.get = mockHttpRequestFunction;
+                mockSuccess();
+            }
+
+            function mockSuccess() {
+                mockHttpRequestFunction
+                    .andReturn(Promise.resolve().then(function () {
+                        returned = true;
+                    }));
+            }
+
+            function mockError() {
+                mockHttpRequestFunction
+                    .andReturn(Promise.reject().then(function () {
+                        returned = true;
+                        //Throw error to ensure chained catch is invoked
+                    }));
+            }
+
+            function httpRequestReturned() {
+                return returned;
+            }
+
+            function mockInterval(interval) {
```

Comment:
We are frugal developers, we speak the jasmine language, we should not introduce new words into the language when they are not necessary because then we have to teach those words to everyone.  Especially when the function name doesn't even match what it does (trigger an interval).